src/serac/physics/base_physics.cpp

```diff
@@ -151,6 +151,18 @@ void BasePhysics::outputStateToDisk(std::optional<std::string> paraview_output_d
         max_order_in_fields = std::max(max_order_in_fields, state->space().GetOrder(0));
       }
 
+      for (const FiniteElementDual* dual : duals_) {
+        // These are really const calls, but MFEM doesn't label them as such
+        serac::FiniteElementDual* non_const_dual = const_cast<serac::FiniteElementDual*>(dual);
+
+        paraview_dual_grid_functions_[dual->name()] =
+            std::make_unique<mfem::ParGridFunction>(const_cast<mfem::ParFiniteElementSpace*>(&dual->space()));
+        non_const_dual->space().GetRestrictionMatrix()->MultTranspose(*dual,
+                                                                      *paraview_dual_grid_functions_[dual->name()]);
+        max_order_in_fields = std::max(max_order_in_fields, dual->space().GetOrder(0));
+        paraview_dc_->RegisterField(dual->name(), paraview_dual_grid_functions_[dual->name()].get());
+      }
+
```

Comment:
Added code looks good. However, this whole block in the `if (!paraview_dc_)` block is getting long, and hence, while the logic in here is pretty simple, it's hard to follow at a glance. Consider folding all of this into a function like `CreateParaViewDataCollection` or something.